src/client/entity-editor/validators/base.js

```diff
@@ -99,6 +99,9 @@ export function validateBoolean(
 export function validateDate(
 	value: mixed, required: boolean = false
 ): boolean {
+	if (!value) {
```

Comment:
Looking at the code, maybe this block should be after the call to `absentAndRequired`?